src/aliceVision/sfm/liealgebra.hpp

```diff
@@ -198,6 +198,77 @@ namespace SE3 {
 
 using Matrix = Eigen::Matrix<double, 4, 4, Eigen::RowMajor>;
 
+/**
+ * Compute the inverse of source matrix
+ */
+inline Eigen::Matrix4d inverse(const Matrix & source)
+{
+  Eigen::Matrix4d ret = Eigen::Matrix4d::Identity();
+  
+  const Eigen::Matrix3d & R = source.block<3, 3>(0, 0);
+  const Eigen::Vector3d & t = source.block<3, 1>(0, 3);
+
+  ret.block<3, 3>(0, 0) = R.transpose();
+  ret.block<3, 1>(0, 3) = - R.transpose() * t;
+
+  return ret;
+}
+
+constexpr int rowFromCoordinates(int row, int col)
+{
+  return col * 4 + row;
+}
+
+/**
+ * Compute the inverse of source matrix
+ */
+inline Eigen::Matrix<double, 16, 16> d_inverse(const Matrix & source)
```

Comment:
```suggestion
inline Eigen::Matrix<double, 16, 16> d_inverse(const Matrix4d & source)
```
If the size of the matrix (and the type) is to be known it's better to specify it